src/sage/sets/primes.py

```diff
@@ -1,187 +1,851 @@
+
+        and that the intersection is empty::
+
+            sage: P.intersection(Q)
+            Empty set of prime numbers
+
+        .. SEEALSO::
+
+            :meth:`intersection`, :meth:`union`
+        """
+        modulus = self._modulus
+        classes = [c for c in range(modulus)
+                   if c % self._modulus not in self._classes]
+        exceptions = {c: not v for c, v in self._exceptions.items()}
+        return Primes(modulus, classes, exceptions)
+
+    def intersection(self, other):
+        r"""
+        Return the intesection of this set with ``other``.
+
+        INPUT:
+
+        - ``other`` -- a subset of the set of prime numbers
+
+        EXAMPLES::
+
+            sage: P = Primes(modulus=5); P
+            Set of prime numbers congruent to 1 modulo 5: 11, 31, 41, 61, ...
+            sage: Q = Primes(modulus=3, classes=[2]); Q
+            Set of prime numbers congruent to 2 modulo 3: 2, 5, 11, 17, ...
+            sage: P.intersection(Q)
+            Set of prime numbers congruent to 11 modulo 15: 11, 41, 71, 101, ...
+
+        TESTS::
+
+            sage: P.intersection(ZZ) == P
+            True
+            sage: P.intersection(RR)
+            Traceback (most recent call last):
+            ...
+            NotImplementedError: boolean operations are only implemented with other sets of prime numbers
+
+        .. SEEALSO::
+
+            :meth:`complement_in_primes`, :meth:`union`
+        """
+        if other is ZZ:
+            return self
+        if not isinstance(other, Primes):
+            raise NotImplementedError("boolean operations are only implemented with other sets of prime numbers")
+        modulus = self._modulus.lcm(other._modulus)
+        classes = [c for c in range(modulus)
+                   if (c % self._modulus in self._classes
+                   and c % other._modulus in other._classes)]
+        exceptions = {}
+        for c, v in self._exceptions.items():
+            if v and c in other:
+                exceptions[c] = True
+            if not v:
+                exceptions[c] = False
+        for c, v in other._exceptions.items():
+            if v and c in self:
+                exceptions[c] = True
+            if not v:
+                exceptions[c] = False
+        return Primes(modulus, classes, exceptions)
+
+    def union(self, other):
+        r"""
+        Return the intesection of this set with ``other``.
+
+        INPUT:
+
+        - ``other`` -- a subset of the set of prime numbers
+
+        EXAMPLES::
+
+            sage: P = Primes(modulus=5); P
+            Set of prime numbers congruent to 1 modulo 5: 11, 31, 41, 61, ...
+            sage: Q = Primes(modulus=3, classes=[2]); Q
+            Set of prime numbers congruent to 2 modulo 3: 2, 5, 11, 17, ...
+            sage: P.union(Q)
+            Set of prime numbers congruent to 1, 2, 8, 11, 14 modulo 15 with 5 included: 2, 5, 11, 17, ...
+
+        TESTS::
+
```

Comment:
A test for the exception logic:
```suggestion
            sage: P = Primes(modulus=5, exceptions={5: True, 11: False}); P
            Set of prime numbers congruent to 1 modulo 5 with 5 included and 11 excluded: 5, 31, 41, 61, ...
            sage: Q = Primes(modulus=4, exceptions={13: False, 11: True}); Q
            Set of prime numbers congruent to 1 modulo 4 with 11 included and 13 excluded: 5, 11, 17, 29, ...
            sage: P.union(Q)
            Set of prime numbers congruent to 1, 9, 11, 13, 17 modulo 20 with 5 included and 13 excluded: 5, 11, 17, 29, ...

```